pkg/manager/volumes/pod_vol_modifier.go

```diff
@@ -0,0 +1,501 @@
+	if pvc.Annotations == nil {
+		pvc.Annotations = map[string]string{}
+	}
+
+	pvc.Annotations[annoKeyPVCLastTransitionTimestamp] = metav1.Now().Format(time.RFC3339)
+}
+
+func (p *podVolModifier) modifyPVCAnnoSpecLastTransitionTimestamp(ctx context.Context, vol *ActualVolume) error {
+	pvc := vol.PVC.DeepCopy()
+	setLastTransitionTimestamp(pvc)
+	updated, err := p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	vol.PVC = updated
+
+	return nil
+
+}
+
+// upgrade revision and snapshot the expected storageclass and size of volume
+func (p *podVolModifier) modifyPVCAnnoSpec(ctx context.Context, vol *ActualVolume, shouldEvict bool) error {
+	pvc := vol.PVC.DeepCopy()
+
+	size := vol.Desired.Size
+	scName := ""
+	if vol.Desired.StorageClass != nil {
+		scName = vol.Desired.StorageClass.Name
+	}
+
+	isChanged := snapshotStorageClassAndSize(pvc, scName, size)
+	if isChanged {
+		upgradeRevision(pvc)
+	}
+
+	if !shouldEvict {
+		setLastTransitionTimestamp(pvc)
+	}
+
+	updated, err := p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	vol.PVC = updated
+
+	return nil
+}
+
+func (p *podVolModifier) syncPVCSize(ctx context.Context, vol *ActualVolume) (bool, error) {
+	capacity := vol.PVC.Status.Capacity.Storage()
+	requestSize := vol.PVC.Spec.Resources.Requests.Storage()
+	if requestSize.String() == vol.Desired.Size && capacity.String() == vol.Desired.Size {
+		return true, nil
+	}
+	if requestSize.String() == vol.Desired.Size {
+		return false, nil
+	}
+
+	pvc := vol.PVC.DeepCopy()
+	q, err := resource.ParseQuantity(vol.Desired.Size)
+	if err != nil {
+		return false, err
+	}
+	pvc.Spec.Resources.Requests[corev1.ResourceStorage] = q
+
+	updated, err := p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	vol.PVC = updated
+
+	return false, nil
+}
+
+func (p *podVolModifier) modifyPVCAnnoStatus(ctx context.Context, vol *ActualVolume) error {
+	pvc := vol.PVC.DeepCopy()
+
+	if pvc.Annotations == nil {
+		pvc.Annotations = map[string]string{}
+	}
+
+	pvc.Annotations[annoKeyPVCStatusRevision] = pvc.Annotations[annoKeyPVCSpecRevision]
+	pvc.Annotations[annoKeyPVCStatusStorageClass] = pvc.Annotations[annoKeyPVCSpecStorageClass]
+	pvc.Annotations[annoKeyPVCStatusStorageSize] = pvc.Annotations[annoKeyPVCSpecStorageSize]
+
+	updated, err := p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	vol.PVC = updated
+
+	return nil
+}
+
+func (p *podVolModifier) modifyVolume(ctx context.Context, vol *ActualVolume) (bool, error) {
+	m := p.getVolumeModifier(vol.Desired.StorageClass)
+	if m == nil {
+		// skip modifying volume by delegation.VolumeModifier
```

Comment:
log here?

Reply:
It's not an invalid path. Size of some volumes can be modified even though delegation.VolumeModifier is not implemented.